File: cpp/71_ordered_fractions.cpp
#include <iostream>
#include <cmath>

int ordered_fractions(int a, int b, int c, int d){
    // calculate mediant (a + c)/(b + d) of two adjacent fractions a/b and c/d
    // updating left fraction with mediant fraction
    // until next mediant has denominator exceeding 1,000,000
    
    // a/b, (a + c)/(b + d), (a + 2c)/(b + 2d), ...
    // b + n*d <= 1,000,000
    // n = floor((1,000,000 - b)/d)
    
    int n = floor((1000000 - b) / d);
    return a + n*c;
}

int main(){
    std::cout << ordered_fractions(2, 5, 3, 7) << "\n";
}

File: c#/_71_OrderedFractions.cs
using System;

public class _71_OrderedFractions{
	public static int Euler71(){
		// calculate mediant of a/b and c/d, (a + c) / (b + d)
		int max = 1_000_000;
		int a = 2, b = 5, c = 3, d = 7;
		// while (b + d <= 1_000_000){
		//     a += c;
		//     b += d;
		// }
		// return a;
		double k = (max - b) / d;
		return a + (int) Math.Floor(k) * c;
	}
	
	public static void Main(){
		Console.WriteLine(Euler71());
	}
}
